mathlibtools/lib.py

```diff
@@ -108,6 +108,8 @@ def download(url: str, target: Path) -> None:
         req = requests.get(url, stream=True)
     except ConnectionError:
         raise LeanDownloadError("Can't connect to " + url)
+    if req.headers['Content-Type'] != 'application/gzip':
```

Comment:
```suggestion
    req.raise_for_status()
```

https://requests.readthedocs.io/en/master/api/#requests.Response.raise_for_status

Reply:
Or maybe `if req.status_code != 200:`, but checking the mime-type seems really fragile...

Reply:
It's a nice trick but the suggestion itself doesn't really works with surrounding code, so I modified it.